src/create-model.jl

```diff
@@ -269,53 +331,66 @@ function add_expression_terms_inter_rp_constraints!(
-            sub_df_flows = filter(
-                [:from, :rep_period] =>
-                    (from, rp) -> from == row_inter.asset && rp == row_map.rep_period,
-                df_flows;
-                view = true,
-            )
-            sub_df_flows.duration = length.(sub_df_flows.timesteps_block)
-            if is_storage_level
-                row_inter.outgoing_flow +=
-                    LinearAlgebra.dot(
-                        sub_df_flows.flow,
-                        sub_df_flows.duration ./ sub_df_flows.efficiency,
-                    ) * row_map.weight
-            else
-                row_inter.outgoing_flow +=
-                    LinearAlgebra.dot(sub_df_flows.flow, sub_df_flows.duration) * row_map.weight
-            end
-
-            if is_storage_level
+        sub_df_map = Dict(
+            year => filter(:period => in(row_inter.periods_block), df_map; view = true) for
+            year in years
+        )
+        for year in years
+            for row_map in eachrow(sub_df_map[year])
                 sub_df_flows = filter(
-                    [:to, :rep_period] =>
-                        (to, rp) -> to == row_inter.asset && rp == row_map.rep_period,
+                    [:from, :rep_period, :year] =>
+                        (from, rp, y) ->
+                            from == row_inter.asset &&
+                                rp == row_map.rep_period &&
+                                y == row_map.year,
                     df_flows;
                     view = true,
                 )
+
                 sub_df_flows.duration = length.(sub_df_flows.timesteps_block)
-                row_inter.incoming_flow +=
-                    LinearAlgebra.dot(
-                        sub_df_flows.flow,
-                        sub_df_flows.duration .* sub_df_flows.efficiency,
-                    ) * row_map.weight
-
-                row_inter.inflows_profile_aggregation +=
-                    profile_aggregation(
-                        sum,
-                        graph[row_inter.asset].rep_periods_profiles,
-                        ("inflows", row_map.rep_period),
-                        representative_periods[row_map.rep_period].timesteps,
-                        0.0,
-                    ) *
-                    graph[row_inter.asset].storage_inflows *
-                    row_map.weight
+                if is_storage_level
+                    row_inter.outgoing_flow +=
+                        LinearAlgebra.dot(
+                            sub_df_flows.flow,
+                            sub_df_flows.duration ./ sub_df_flows.efficiency,
+                        ) * row_map.weight
+                else
+                    row_inter.outgoing_flow +=
+                        LinearAlgebra.dot(sub_df_flows.flow, sub_df_flows.duration) * row_map.weight
+                end
+
+                if !is_storage_level # got an error here, that's why I modified this condition
```

Comment:
@datejada This condition was `if is_storage_level`, which is the same as the previous one, is this intentional or a glitch?

Reply:
It is not a glitch, but the function is confusing, so we will clean it in the refactor. Thanks for the feedback.